src/screens/login/index.tsx

```diff
@@ -2,64 +2,98 @@ import React, { FC, useState } from 'react';
 
 import { useNavigation } from '@react-navigation/native';
 import { NativeStackNavigationProp } from '@react-navigation/native-stack';
+import * as EmailValidator from 'email-validator';
 
 import { login, registration } from '../../api/userAPI';
 
 import Background from '../../components/background/Background';
 import Title from '../../components/title/Title';
+import { ErrorWindow } from './components/errorWindow';
 
+import { ERROR_MESSAGE } from '../../../errors/errors';
 import { Icons } from '../../assets/Icons';
 import { Colors } from '../../constants/colors';
 import { TemplateType } from '../../constants/templateType';
+import Text from '../../constants/text';
 import { StackParams, UserType } from '../../types/types';
 
-import { ButtonText, LoginFrame, LoginInput, SignInButton, SignUpButton } from './styles';
+import { ButtonText, ErrorLogin, LoginFrame, LoginInput, SignInButton, SignUpButton, TextError } from './styles';
 
 const Login: FC = () => {
   const navigation = useNavigation<NativeStackNavigationProp<StackParams>>();
   const [email, setEmail] = useState<string | undefined>();
   const [password, setPassword] =useState<string | undefined>();
   const [user, setUser] = useState<UserType | undefined>();
+  const [errorMessage, setErrorMessage] = useState<string | undefined>(undefined);
 
   const signIn = async (): Promise<void> => {
-    const userData = await login(email, password);
-    if (userData) {
-      setUser(userData);
-      navigation.navigate('Main');
+    try {
+      const userData = await login(email, password);
+      if (typeof userData === 'number') {
+        setErrorMessage(ERROR_MESSAGE[userData]);
+      }
+      if ((userData as UserType).email) {
```

Comment:
if your request was successful it will always return userData so you don't need to add if here. If your request returned error it will always go to catch block